### ISSUE TITLE
[CI] CacheFile.decrementRefCount() trips an assertion error on Windows

### ISSUE THREAD

**Issue by astefan** [elastic/elasticsearch#67579]
**Build scan**: https://gradle-enterprise.elastic.co/s/o436j2lhw4s6k

**Repro line**: `gradlew ':x-pack:plugin:searchable-snapshots:internalClusterTest' --tests "org.elasticsearch.xpack.searchablesnapshots.RetrySearchIntegTests.testRetryPointInTime" -Dtests.seed=35046589742E5834 -Dtests.security.manager=true -Dtests.locale=it -Dtests.timezone=AET -Druntime.java=11`

**Reproduces locally?**: No (tested in Windows 10)

**Applicable branches**: master

**Failure excerpt**:
```
org.elasticsearch.xpack.searchablesnapshots.RetrySearchIntegTests > testRetryPointInTime FAILED
    com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=427, name=elasticsearch[node_s3][generic][T#2], state=RUNNABLE, group=TGRP-RetrySearchIntegTests]
        at __randomizedtesting.SeedInfo.seed([35046589742E5834:407064FEFFD2E71C]:0)

        Caused by:
        java.lang.AssertionError
            at __randomizedtesting.SeedInfo.seed([35046589742E5834]:0)
            at org.elasticsearch.index.store.cache.CacheFile.decrementRefCount(CacheFile.java:231)
            at org.elasticsearch.index.store.cache.CacheFile.startEviction(CacheFile.java:243)
            at org.elasticsearch.core.internal.io.IOUtils.close(IOUtils.java:74)
            at org.elasticsearch.core.internal.io.IOUtils.closeWhileHandlingException(IOUtils.java:164)
            at org.elasticsearch.xpack.searchablesnapshots.cache.CacheService.onCacheFileRemoval(CacheService.java:518)
            at org.elasticsearch.xpack.searchablesnapshots.cache.CacheService.lambda$new$1(CacheService.java:170)
            at org.elasticsearch.common.cache.Cache.delete(Cache.java:799)
            at org.elasticsearch.common.cache.Cache.lambda$new$6(Cache.java:499)
            at org.elasticsearch.common.cache.Cache$CacheSegment.remove(Cache.java:316)
            at org.elasticsearch.common.cache.Cache.invalidate(Cache.java:529)
            at org.elasticsearch.xpack.searchablesnapshots.cache.CacheService.processShardEviction(CacheService.java:429)
            at org.elasticsearch.xpack.searchablesnapshots.cache.CacheService$1.doRun(CacheService.java:365)
            at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
            at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
            at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
            at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:680)
            at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
            at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
            at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by astefan** [comment]
Another one [here](https://gradle-enterprise.elastic.co/s/2wp5o4iqetf7m). Identical stack trace, different test class.

`gradlew ':x-pack:plugin:searchable-snapshots:internalClusterTest' --tests "org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests.testSnapshotMountedIndexLeavesBlobsUntouched" -Dtests.seed=52ACA4AAA8360B0 -Dtests.security.manager=true -Dtests.locale=und -Dtests.timezone=America/Buenos_Aires -Druntime.java=8`

```
org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests > testSnapshotMountedIndexLeavesBlobsUntouched FAILED
    com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=1221, name=elasticsearch[node_s0][generic][T#10], state=RUNNABLE, group=TGRP-SearchableSnapshotsIntegTests]
        at __randomizedtesting.SeedInfo.seed([52ACA4AAA8360B0:E358136A5ED04B0A]:0)

        Caused by:
        java.lang.AssertionError
            at __randomizedtesting.SeedInfo.seed([52ACA4AAA8360B0]:0)
            at org.elasticsearch.index.store.cache.CacheFile.decrementRefCount(CacheFile.java:231)
            at org.elasticsearch.index.store.cache.CacheFile.startEviction(CacheFile.java:243)
            at org.elasticsearch.core.internal.io.IOUtils.close(IOUtils.java:74)
            at org.elasticsearch.core.internal.io.IOUtils.closeWhileHandlingException(IOUtils.java:164)
            at org.elasticsearch.xpack.searchablesnapshots.cache.CacheService.onCacheFileRemoval(CacheService.java:518)
            at org.elasticsearch.xpack.searchablesnapshots.cache.CacheService.lambda$new$1(CacheService.java:170)
            at org.elasticsearch.common.cache.Cache.delete(Cache.java:798)
            at org.elasticsearch.common.cache.Cache.lambda$new$6(Cache.java:498)
            at org.elasticsearch.common.cache.Cache$CacheSegment.remove(Cache.java:316)
            at org.elasticsearch.common.cache.Cache.invalidate(Cache.java:528)
            at org.elasticsearch.xpack.searchablesnapshots.cache.CacheService.processShardEviction(CacheService.java:429)
            at org.elasticsearch.xpack.searchablesnapshots.cache.CacheService.access$100(CacheService.java:67)
            at org.elasticsearch.xpack.searchablesnapshots.cache.CacheService$1.doRun(CacheService.java:365)
            at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
            at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
            at java.util.concurrent.FutureTask.run(FutureTask.java:266)
            at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:684)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
            at java.lang.Thread.run(Thread.java:748)
```

**Comment by tlrx** [comment]
Thanks @astefan. I added a more descriptive message to this assertion in #67561, hopefully this will confirm what I suspect. 

I'd like to have more failures to see if that only happens on Windows.

**Comment by hendrikmuhs** [comment]
New failure: https://gradle-enterprise.elastic.co/s/whenryymqi4nm

**Comment by tlrx** [comment]
Thanks @hendrikmuhs. I just opened #67633 to adapt the failing assertion.

**Comment by tlrx** [comment]
Closed by #67633
